### ISSUE TITLE
amLoggedAs causes error in Lumen

### ISSUE THREAD

**Issue by a300600st** [Codeception/Codeception#3340]
```
[PHPUnit_Framework_Exception] call_user_func_array() expects parameter 1 to be a valid callback,
    class 'Illuminate\Auth\RequestGuard' does not have a method 'driver'
```
When I try to use amLoggedAs. Changing the function from
```
public function amLoggedAs($user, $driver = null)
{
    if ($user instanceof Authenticatable) {
        $this->app['auth']->driver($driver)->setUser($user);
    } else {
        $this->app['auth']->driver($driver)->attempt($user);
    }
}
```
to
```
public function amLoggedAs($user, $driver = null)
{
    if ($user instanceof Authenticatable) {
        $this->app['auth']->guard($driver)->setUser($user);
    } else {
        $this->app['auth']->guard($driver)->attempt($user);
    }
}
```
fixes the error but I don't have enough experience to know the full ramifications of this change.

**Comment by Naktibalda** [comment]
It looks consistent with https://github.com/Codeception/Codeception/commit/192e3339fa259d53b0e7caa7a1afe5b2767b55fb change in Laravel5 module.

@a300600st Could you copy that change from Laravel5 to Lumen and make a pull request?

@janhenkgerritsen Do you support Lumen module?

**Comment by a300600st** [comment]
I messed up the first pull request but I think the second one is correct.

**Comment by janhenkgerritsen** [comment]
Thanks for the pull request! 

The Lumen module probably needs some more work for version 5.2, because there were some major changes between Lumen 5.1 and Lumen 5.2. I haven't had the time to do this though, hopefully soon.

**Comment by a300600st** [comment]
Good to know. We just switched to codeception with lumen 5.2 so if I find other problems I'll see what I can do to help fix them.

**Comment by janhenkgerritsen** [comment]
@a300600st I recently finished a large refactoring of the Lumen module, but it probably needs a bit more testing in real applications before I merge it into the Codeception 2.2 branch. Since I know you are using the Lumen module, would it be possible for you to test out the refactored module and see if you run into  any issues?

If so, take a look at [this comment](https://github.com/Codeception/Codeception/issues/3494#issuecomment-246591781) I made on another issue for more details.
